upgrade/v110to120/upgrade.go

```diff
@@ -179,74 +209,93 @@ func translateStorageClassRecurringJobsToSelector(namespace string, kubeClient *
 			recurringJobIDs = append(recurringJobIDs, id)
 		}
 	}
-	log.Debugf(upgradeLogPrefix+"Converting recurringJobs %v to recurringJobSelector", recurringJobIDs)
-	scRecurringJobSelectors := []recurringJobSelector{}
-	scRecurringJobSelectorJSON, ok := sc.Parameters["recurringJobSelector"]
-	scRecurringJobSelectorIDs := []string{}
+	if err := run.createRecurringJobCRs(); err != nil {
+		return err
+	}
+	if err := run.convertToSelectors(recurringJobIDs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (run *recurringJobUpgrade) convertToSelectors(recurringJobIDs []string) (err error) {
+	revertLog := run.log
+	defer func() {
+		run.log = revertLog
+		err = errors.Wrapf(err, "convert storageClass recurringJobs to recurringJobSelector failed")
+	}()
+	run.log = run.log.WithField("action", "convert-recurring-jobs-to-selectors")
+
+	if len(recurringJobIDs) == 0 {
+		run.log.Debug(upgradeLogPrefix + "Found 0 recurring job to convert to recurring job selector")
+		return
+	}
+
+	selectors := []recurringJobSelector{}
```

Comment:
Can you directly use a map instead? Then you don't need to store all selector ID in `selectorIDs` then rely on `selectorIDs` to skip the duplication in `recurringJobIDs`.

This change will improve the readability of the funcion. The only extra operation is that this map (values) should be converted to a list in the end.

Reply:
I prefer not to do the extra conversion because
`run.storageClass.Parameters["recurringJobSelector"]` comes as an array. To use map instead will need to convert the existing recurringJobSelector to map and then at the end back to array again.

Reply:
Got it.